### ISSUE TITLE
Connection pool's timer thread don't stop on pool close

### ISSUE THREAD

**Issue by golubovai** [oracle/python-oracledb#361]
1. What versions are you using?
Oracle Database 19c

2. Is it an error or a hang or a crash?
It is a hang.

3. What error(s) or behavior you are seeing?
A running timer thread after connection pool close prevents the application from graceful shutdown (e.g. FastApi).

4. Does your application call init_oracle_client()?
No, it's Thin mode.

5. Include a runnable Python script that shows the problem.
It's hard to write script, describing a problem, but I think that it's obvious.
As a workround you can stop this thread manually after closing the pool. Thank goodness the timer thread has a cancel method:
``
    for timer in [thread for thread in threading.enumerate() if isinstance(thread, threading.Timer)]:
        timer.cancel()
``

**Comment by cjbj** [comment]
Thanks for the report. What's your python-oracledb version.  If it's not 2.2.1, then can you try with that version?

Also, if it's hard to write a script to reproduce, but can you describe anything else about your pool sizing?  What OS?

**Comment by golubovai** [comment]
Python 3.12, oracledb 2.2.1, indirect test case here:

```
import oracledb
import threading

oracledb.defaults.config_dir = ' '
u = ' '
p = ' '
dsn = ' '

def has_timer_thread():
    for _ in [thread for thread in threading.enumerate() if isinstance(thread, threading.Timer)]:
        return True
    return False

def c(u, p, dsn):
    print('start')
    pool = oracledb.create_pool(user=u,
                                password=p,
                                dsn=dsn,
                                timeout=20,
                                min=0,
                                max=10)
    print(f'before acquire: {has_timer_thread()}')
    connection = pool.acquire()
    print(f'after acquire: {has_timer_thread()}')
    connection.close()
    print(f'after connection close: {has_timer_thread()}')
    pool.close()
    print(f'after pool close: {has_timer_thread()}')
    print('end')


def main():
    c(u, p, dsn)


if __name__ == '__main__':
    main()
```

**Comment by anthony-tuininga** [comment]
This issue was addressed in version 2.3.0 which was released last week.
